### PR TITLE
[Audit fix] remove address==0 check and move errors up

### PR DESCRIPTION
**Pull request by alpinechicken**
# Task:

- Remove address==0 check 
- Moved errors up
- Removed unused error code

## Description

Crabv2 address could previously only be set once. Now crabv2 address can be reset by owner.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested

Tests pass.

## FE Checklist

- [ x] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] My changes generate no new warnings
- [x ] Added video recordings if it is a UI change

## User Facing Checklist

- [x ] I fully understand the user problem this PR is solving
- [x ] I know who the target user is for this PR and have a deep understanding of that user
- [ x] I have tried this flow thinking from the pov of the target user for this PR
- [ ] I (or working w someone on team) have scheduled a user test for this PR (if it is a large change)

### CODE REVIEW

**Comment by aparnakr** [comment]
@aleone you also need to fix the tests :(

**Comment by sanandnarayan** [comment]
Duplicate work with
https://github.com/opynfinance/squeeth-monorepo/pull/563 ?

**Review by aparnakr — APPROVED** [review]

File: packages/hardhat/contracts/strategy/CrabMigration.sol [hunk 1]
```diff
@@ -83,17 +81,17 @@ contract CrabMigration is Ownable {
...
 
     modifier afterMigration() {
-        require(isMigrated, "M3");
+        require(isMigrated, "M2");
         _;
     }
 
     modifier afterInitialized() {
-        require(address(crabV2) != address(0), "M8");
+        require(address(crabV2) != address(0), "M7");
```
Comment (aleone) [inline packages/hardhat/contracts/strategy/CrabMigration.sol:94]: This the wrong error code (and was wrong before).  We need to keep the M10 error code from before.

File: packages/hardhat/contracts/strategy/CrabMigration.sol [hunk 2]
```diff
@@ -83,17 +81,17 @@ contract CrabMigration is Ownable {
...
     }
 
     modifier afterMigration() {
-        require(isMigrated, "M3");
+        require(isMigrated, "M2");
         _;
     }
 
     modifier afterInitialized() {
-        require(address(crabV2) != address(0), "M8");
```
Comment (aleone) [inline packages/hardhat/contracts/strategy/CrabMigration.sol:96]: this isnt M8

**Review by aleone — CHANGES_REQUESTED** [review]

File: packages/hardhat/contracts/strategy/CrabMigration.sol (hunk 1 above)
Comment (alpinechicken) [reply packages/hardhat/contracts/strategy/CrabMigration.sol:94]: added

**Review by aleone — APPROVED** [review]

**Pull request merged**